### ISSUE TITLE
Staging Review finding: Copy action isn't announced to screen reader users

### ISSUE THREAD

**Issue by shiragoodman** [department-of-veterans-affairs/va.gov-team#93693]
_Need help? Please review [how to read a Staging Review ticket](https://depo-platform-documentation.scrollhelp.site/collaboration-cycle/anatomy-of-a-staging-review-issue-ticket). Tag `@platform-governance-team-members` on Slack if you need further assistance._

## Product Information
**Team:** Transition Experience
**Product:** Transition Experience MVP v1
**Feature:** V1 Transition Experience

## Findings details
**VA.gov Experience Standard - issue:** [User experiences an unexpected interaction with a component or pattern.](https://design.va.gov/about/experience-standards/#user-experiences-predictable-interactions-with-components)
**VA.gov Experience Standard - category:** [Usability](https://design.va.gov/about/experience-standards/#category-usability)
**Launch-blocking:** Yes
**Design System review:** No
**Collab Cycle Reviewer:** @briandeconinck (Accessibility)

### Description

In the "Save your results" modal, pressing the "Copy" button results in no announcement to screen reader users. Although a success message does appear on screen, that message is not announced to screen reader users. Nothing is announced at all after pressing "Copy," leaving users who can't see the visual changes to the content of the modal to guess about what happened. Did the button work? Is it broken? Did they accidentally press the wrong key? Should they try again?

Generally speaking, it's bad form to force users to guess about whether or not the interface is broken or working.

<details><summary>Link, screenshot or steps to recreate</summary>

![Save your results modal, with a success message reading "URL has been copied to your clipboard" and a visible focus indicator still around the Copy button.](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/platform/working-with-vsp/vsp-collaboration-cycle/staging-review-images/82863-Brian-3.png?raw=true)

Steps to reproduce:
1. Navigate to the results page of the product.
2. Activate a screen reader. (Easiest options for testing: VoiceOver on Mac or NVDA on Windows.)
3. Press [tab] until focus moves to the "Save your results" button.
4. Press [enter] to open the modal.
5. Press [tab] to move focus to the "Copy" button.
6. Press [enter] to activate the "Copy" button.
7. Note that your screen reader does not announce anything after you press "Copy."

</details>

## Recommended action

There are a variety of options here, but what we typically do on VA.gov is move keyboard focus to the success alert (using `tabindex="-1"` to make the alert focusable) as part of the action triggered by the "Copy" button.

For a screen reader user, this will result in the success message immediately being read, providing instant feedback to verify that the action occurred successfully. (And likewise, if there is any error state, focus should move to that error.)

## References

- **Accessibility Defect Severity:** [1: Critical. Must be fixed before launch.](https://depo-platform-documentation.scrollhelp.site/developer-docs/accessibility-defect-severity-rubric#Accessibilitydefectseverityrubric-a11y-defect-1)
- **WCAG Success Criteria:** N/A
- **Modality:** Screen reader
- **Design System Component:** N/A
- **Design System Pattern or Template:** N/A
- **Design System Foundation:** N/A
- **Content Style Guide:** N/A
- **Context:** N/A

---

### Next Steps for the VFS Team

- [ ] **Questions?** For the most timely response, comment on Slack in your team channel tagging `@platform-governance-team-members` with any questions or to get help validating the issue.
- [ ] **Ticket ownership:** If you believe that this issue is out of scope, not your team's responsibility, or a Design System issue, comment and tag `@platform-governance-team-members` on your team channel in Slack to provide an explanation and who you believe is responsible. The Governance team will follow up.
- [ ] **Close this ticket** when the issue has been resolved or validated by your Product Owner.

**Comment by nkontrabecki** [comment]
Confirmed fixed and tested in Sprint 13 on 10/7/2024.
